packages/quantic/force-app/main/default/lwc/headlessLoader/headlessLoader.js

```diff
@@ -0,0 +1,145 @@
+import HeadlessPath from '@salesforce/resourceUrl/coveoheadless';
+import AtomicPath from '@salesforce/resourceUrl/atomicutils';
+// @ts-ignore	
+import { loadScript } from 'lightning/platformResourceLoader';
+
+const DEBOUNCE_DELAY = 200;
+let timeout;
+
+/**
+ * Debounces a function execution.
+ * @param {Function} func The function for which to delay execution.
+ * @param {Number} wait The time to delay in milliseconds.
+ */
+const debounce = (func, wait) => {
+  return function executedFunction(...args) {
+    const later = () => {
+      clearTimeout(timeout);
+      func(...args);
+    };
+    clearTimeout(timeout);
+    // eslint-disable-next-line @lwc/lwc/no-async-operation
+    timeout = setTimeout(later, wait);
+  };
+};
+
+/**
+ * Cancels the delayed search query.
+ */
+const cancelInitialSearch = () => {
+  if (timeout) {
+    clearTimeout(timeout);
+    timeout = undefined;
+  }
+}
+
+/**
+ * Dispatches search request.
+ */
+const executeInitialSearch = debounce(() => {
+  window.coveoHeadless.engine.dispatch(
+    CoveoHeadless.SearchActions.executeSearch(
+      CoveoHeadless.AnalyticsActions.logInterfaceLoad()
+    )
+  );
+}, DEBOUNCE_DELAY);
+
+/**
+ * Returns true if registered components are initialized, false otherwise.
+ */
+const getAreComponentsReady = () => !window.coveoHeadless.components.find(component => component.initialized === false);
+
+/**
+ * Loads dependencies and returns an initialized Headless engine. 
+ * @param element 
+ */
+async function initEngine(element) {
+  let engine;
+  try {
+    await loadScript(element, HeadlessPath + '/browser/headless.js');
+    await loadScript(element, AtomicPath + '/atomic-utils.js');
+  
+    const config = CoveoHeadless.HeadlessEngine.getSampleConfiguration();
+  
+    engine = new CoveoHeadless.HeadlessEngine({
+      configuration: config,
+      reducers: CoveoHeadless.searchAppReducers,
+    });
+  } catch (error) {
+    console.error('Fatal error: unable to initialize Coveo Headless', error);
+  }
+  return engine;
+}
+
+/**
+ * Registers a component for future initialization.
+ * @param element 
+ */
+function registerComponentForInit(element) {
+  cancelInitialSearch();
+  if (!window.coveoHeadless) {
+    window.coveoHeadless = {
+      components: [],
+      engine: undefined
+    }
+  }
+  const isComponentRegistered = window.coveoHeadless.components.find((component) => component.element === element)
```

Comment:
this snipet is also used on line 100.
Maybe you can create a small private for it ? 🙂

Reply:
definitely!